File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/PackageSpotModel.java
package com.tourdesign.platform.model;

import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PackageSpotModel {
    private Long id;
    private Integer orderIndex;

    private TravelPackageModel travelPackage;
    private TouristSpotModel touristSpot;
}

File: frontend/tour-design-web/src/app/models/package-spot.model.ts
// src/app/models/package-spot.model.ts
import { TravelPackageModel } from './travel-package.model';
import { TouristSpotModel } from './tourist-spot.model';

export interface PackageSpotModel {
  id: number;
  orderIndex?: number;
  travelPackage?: TravelPackageModel;
  touristSpot?: TouristSpotModel;
}
